### ISSUE TITLE
PHPUnit failures from database table not exists errors on WordPress trunk

### ISSUE THREAD

**Issue by JDGrimes** [WordPoints/wordpoints#727]
The [last build started getting failures on WordPress `develop`](https://travis-ci.org/WordPoints/wordpoints/jobs/278779335). Investigation lead to the discovery that the offending commit in core was https://core.trac.wordpress.org/changeset/41560. I left a [comment on the ticket](https://core.trac.wordpress.org/ticket/39766#comment:14), explaining:

>The use of `plugin_sandbox_scrape()` in `activate_plugin()` is the problem, in particular, the running of the `'plugins_loaded'` action.

A little more digging revealed that the cause of the errors was that WordPoints wasn't getting installed, and that's why the database tables didn't exist. That's because `wordpoints_register_installer()` is hooked to `'plugins_loaded'`, as is the `wordpoints_installables_maybe_update()` function (via `'wordpoints_extensions_loaded'`), which causes the update check to run. In the end this causes the version to be saved in the database, which means that when `wordpoints_activate()` *later* gets called, it doesn't run the install, because it thinks WordPoints is already installed.

And actually, *WordPoints* is. But the points component is not (which is where the errors are coming from). Because we don't install the points component as part of the install routine itself, only in `wordpoints_activate()` *if we are installing the plugin for the first time*. (Interesting to note that in prior versions we actually did do this as part of the install routine itself, which meant that it would have been triggered, whereas now it isn't part of `WordPoints_Installable_Core`.)

Hopefully WordPress changes this before we release, but we'll keep track of things here.

**Comment by JDGrimes** [comment]
That issue has been fixed by just removing `plugins_loaded` from being called there for now, but as the other actions are still called, we are still getting some other errors:

```
There were 2 failures:

1) WordPoints_Extension_Updates_Test::test_fill_not_set

Failed asserting that 1507060045 is identical to 0.

/home/travis/build/WordPoints/wordpoints/tests/phpunit/tests/classes/extension/updates.php:208

2) WordPoints_Extension_Updates_Test::test_fill_empty_already_set

Failed asserting that 1507060045 is identical to 0.

/home/travis/build/WordPoints/wordpoints/tests/phpunit/tests/classes/extension/updates.php:244
```

This is because these tests expect the extension updates transient not to be set, because `wordpoints_check_for_extension_updates()` won't have been called. However, it is hooked to `admin_init`, which is one of the actions still be run on activation now. Hopefully this will be fixed by a fuller revert before WordPress 4.9.0 is released.

**Comment by JDGrimes** [comment]
This was fixed in https://core.trac.wordpress.org/changeset/41819.
